### DISCUSSION TITLE
Definitions of `D_c` and `D_d` in the augmented system

### DISCUSSION THREAD

**Discussion by pbmadabhushi** [coin-or/Ipopt#686] [Q&A] [answered]
I'm curious to know what `D_c` and `D_d` in IPOPT code (see screenshot) map to in the IPOPT implementation paper? Any direction will be helpful.

![image](https://github.com/coin-or/Ipopt/assets/131873102/d5c671ee-3374-4759-81d5-0eff3d368de9)

**Comment by svigerske** [comment, marked as answer]
They do not seem to appear in (13) in the paper.

But as far as I see, they are also set to zero in the implementation: NULL is passed for D_c and D_d in 
https://github.com/coin-or/Ipopt/blob/391b7716ae5e5791282260a3e35d8d8ad2cd2383/src/Algorithm/IpPDFullSpaceSolver.cpp#L520-L522
And NULL means a 0-matrix here.

Maybe there was some thought once that they could be useful.
